presto-hive/src/main/java/com/facebook/presto/hive/parquet/HdfsParquetDataSource.java

```diff
@@ -80,43 +51,27 @@ public void close()
     }
 
     @Override
-    public final void readFully(long position, byte[] buffer)
-    {
-        readFully(position, buffer, 0, buffer.length);
-    }
-
-    @Override
-    public final void readFully(long position, byte[] buffer, int bufferOffset, int bufferLength)
-    {
-        readBytes += bufferLength;
-
-        long start = System.nanoTime();
-        readInternal(position, buffer, bufferOffset, bufferLength);
-        long currentReadTimeNanos = System.nanoTime() - start;
-
-        readTimeNanos += currentReadTimeNanos;
-        stats.readDataBytesPerSecond(bufferLength, currentReadTimeNanos);
-    }
-
-    private void readInternal(long position, byte[] buffer, int bufferOffset, int bufferLength)
+    protected void readInternal(long position, byte[] buffer, int bufferOffset, int bufferLength)
     {
         try {
+            long start = System.nanoTime();
             inputStream.readFully(position, buffer, bufferOffset, bufferLength);
+            stats.readDataBytesPerSecond(bufferLength, System.nanoTime() - start);
         }
         catch (PrestoException e) {
             // just in case there is a Presto wrapper or hook
             throw e;
         }
         catch (Exception e) {
-            throw new PrestoException(HIVE_FILESYSTEM_ERROR, format("Error reading from %s at position %s", id, position), e);
+            throw new PrestoException(HIVE_FILESYSTEM_ERROR, format("Error reading from %s at position %s", getId(), position), e);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: position (declared at line 54). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 66. Consider formatting this data with &#39;%d&#39; instead.